tfjs-backend-wasm/src/kernels/BroadcastArgs.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =============================================================================
+ */
+import {backend_util, BroadcastArgs, BroadcastArgsInputs, KernelConfig, TensorInfo} from '@tensorflow/tfjs-core';
+
+import {BackendWasm} from '../backend_wasm';
+
+export function broadcastArgs(args: {
+  inputs: BroadcastArgsInputs,
+  backend: BackendWasm,
+}): TensorInfo {
+  const {inputs, backend} = args;
+  const {s0, s1} = inputs;
+
+  const s0Vals = backend.typedArrayFromHeap(s0);
+  const s1Vals = backend.typedArrayFromHeap(s1);
+
+  const broadcastShape = backend_util.assertAndGetBroadcastShape(
```

Comment:
Maybe we should have a generic way to fall back to the CPU implementation for ops that aren't implemented in WASM. That would allow us to simply not implement this op, and it would use the fallback by default. IIRC we have something like this for the WebGL backend, but it's mostly used to avoid sending tensors to the GPU for really small ops.

Reply:
I'm thinking about the same thing. I may do it in a separate CL later.
This is not the first wasm kernel runs with cpu util/kernel function, and there was no general bridge for wasm-cpu kernel. I will revisit all the cpu function usages in wasm and find a general solution for them after I am familiar with more kernels.